### PR TITLE
Fix bug in split when program running on UDP contains space in name

### PR DESCRIPTION
**Pull request by HervisDaubeny**
Suggested solution for https://github.com/kellyjonbrazil/jc/issues/446

### CODE REVIEW

**Comment by HervisDaubeny** [comment]
I made a slight mistake when claiming that the programs running on UDP will never have the "State" column filled. According to the documentation - https://linux.die.net/man/8/netstat 
> The state of the socket. Since there are no states in raw mode and usually no states used in UDP, this column may be left blank.

I will adjust my solution and let you know when it will be ready.

**Comment by kellyjonbrazil** [comment]
Thanks for the fix! I'll take a look in the next few days.

**Comment by HervisDaubeny** [comment]
I pushed an update of the solution which should be correct according to the netstat documentation.

**Comment by kellyjonbrazil** [comment]
Thanks for the update - looks like tests failed, though.

**Comment by HervisDaubeny** [comment]
I will look into it, just to know, how do you get the output for the test so I could replicate it? Is it calling a plain `netstat` command?

**Comment by kellyjonbrazil** [comment]
Hi @HervisDaubeny. If you have cloned the repo you should be able to run `./runtests.sh` in the root of the project, which will run all the tests. Alternatively you can run the `test_netstat.py` script directly within the `tests` folder.

The command output and expected JSON files are stored within the `fixtures` folder under `tests`.

I'm not sure if you can see the Github actions test results, but here is what they are showing (these are the same as the local tests):
```
======================================================================
FAIL: test_netstat_l_centos_7_7 (test_netstat.MyTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/jc/jc/tests/test_netstat.py", line 282, in test_netstat_l_centos_7_7
    self.assertEqual(jc.parsers.netstat.parse(self.centos_7_7_netstat_l, quiet=True), self.centos_7_7_netstat_l_json)
AssertionError: Lists differ: [{'pr[1807 chars]te': None, 'kind': 'network', 'local_port': 'i[5199 chars]et'}] != [{'pr[1807 chars]te': '7', 'kind': 'network', 'local_port': 'ip[5156 chars]et'}]

First differing element 7:
{'pro[95 chars]te': None, 'kind': 'network', 'local_port': 'i[124 chars]': 7}
{'pro[95 chars]te': '7', 'kind': 'network', 'local_port': 'ip[81 chars]pv6'}

Diff is 8573 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_netstat_l_ubuntu_18_4 (test_netstat.MyTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/jc/jc/tests/test_netstat.py", line 288, in test_netstat_l_ubuntu_18_4
    self.assertEqual(jc.parsers.netstat.parse(self.ubuntu_18_4_netstat_l, quiet=True), self.ubuntu_18_4_netstat_l_json)
AssertionError: Lists differ: [{'pr[1566 chars]te': None, 'kind': 'network', 'local_port': 'i[3499 chars]et'}] != [{'pr[1566 chars]te': '7', 'kind': 'network', 'local_port': 'ip[3456 chars]et'}]

First differing element 6:
{'pro[95 chars]te': None, 'kind': 'network', 'local_port': 'i[124 chars]': 7}
{'pro[95 chars]te': '7', 'kind': 'network', 'local_port': 'ip[81 chars]pv6'}

Diff is 6213 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_netstat_sudo_aeep_centos_7_7 (test_netstat.MyTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/jc/jc/tests/test_netstat.py", line 318, in test_netstat_sudo_aeep_centos_7_7
    self.assertEqual(jc.parsers.netstat.parse(self.centos_7_7_netstat_sudo_aeep, quiet=True), self.centos_7_7_netstat_sudo_aeep_json)
AssertionError: Lists differ: [{'pr[3009 chars]te': None, 'user': '7', 'inode': None, 'progra[26124 chars]787}] != [{'pr[3009 chars]te': '7', 'user': 'root', 'inode': 20819, 'pro[26122 chars]787}]

First differing element 9:
{'pro[95 chars]te': None, 'user': '7', 'inode': None, 'progra[161 chars]pv6'}
{'pro[95 chars]te': '7', 'user': 'root', 'inode': 20819, 'pro[159 chars]pv6'}

Diff is 35438 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_netstat_sudo_aeep_ubuntu_18_4 (test_netstat.MyTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/jc/jc/tests/test_netstat.py", line 324, in test_netstat_sudo_aeep_ubuntu_18_4
    self.assertEqual(jc.parsers.netstat.parse(self.ubuntu_18_4_netstat_sudo_aeep, quiet=True), self.ubuntu_18_4_netstat_sudo_aeep_json)
AssertionError: Lists differ: [{'pr[2754 chars]te': None, 'user': '7', 'inode': None, 'progra[20137 chars]522}] != [{'pr[2754 chars]te': '7', 'user': 'systemd-network', 'inode': [20146 chars]522}]

First differing element 8:
{'pro[95 chars]te': None, 'user': '7', 'inode': None, 'progra[162 chars]pv6'}
{'pro[95 chars]te': '7', 'user': 'systemd-network', 'inode': [171 chars]pv6'}

Diff is 27794 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_netstat_sudo_lnp_centos_7_7 (test_netstat.MyTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/jc/jc/tests/test_netstat.py", line 306, in test_netstat_sudo_lnp_centos_7_7
    self.assertEqual(jc.parsers.netstat.parse(self.centos_7_7_netstat_sudo_lnp, quiet=True), self.centos_7_7_netstat_sudo_lnp_json)
AssertionError: Lists differ: [{'pr[2154 chars]te': None, 'program_name': 'NetworkManager', '[6525 chars]469}] != [{'pr[2154 chars]te': '7', 'program_name': 'NetworkManager', 'k[6523 chars]469}]

First differing element 7:
{'pro[91 chars]te': None, 'program_name': 'NetworkManager', '[144 chars]: 58}
{'pro[91 chars]te': '7', 'program_name': 'NetworkManager', 'k[142 chars]: 58}

Diff is 10547 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_netstat_sudo_lnp_ubuntu_18_4 (test_netstat.MyTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/jc/jc/tests/test_netstat.py", line 312, in test_netstat_sudo_lnp_ubuntu_18_4
    self.assertEqual(jc.parsers.netstat.parse(self.ubuntu_18_4_netstat_sudo_lnp, quiet=True), self.ubuntu_18_4_netstat_sudo_lnp_json)
AssertionError: Lists differ: [{'pr[1914 chars]te': None, 'program_name': 'systemd-network', [4[29](https://github.com/kellyjonbrazil/jc/actions/runs/6144245405/job/16683078787?pr=447#step:6:30)0 chars]607}] != [{'pr[1914 chars]te': '7', 'program_name': 'systemd-network', '[4288 chars]607}]

First differing element 6:
{'pro[91 chars]te': None, 'program_name': 'systemd-network', [145 chars]: 58}
{'pro[91 chars]te': '7', 'program_name': 'systemd-network', '[143 chars]: 58}

Diff is 7549 characters long. Set self.maxDiff to None to see it.

----------------------------------------------------------------------
Ran 13[31](https://github.com/kellyjonbrazil/jc/actions/runs/6144245405/job/16683078787?pr=447#step:6:32) tests in 10.2[58](https://github.com/kellyjonbrazil/jc/actions/runs/6144245405/job/16683078787?pr=447#step:6:59)s

FAILED (failures=6)
Error: Process completed with exit code 1.
```

**Comment by kellyjonbrazil** [comment]
Looks like I can fix the broken tests by adding one more state to the list: "7". Not sure if there are any other word/numeric states we need t worry about.

File: jc/parsers/netstat_linux.py [hunk 1]
```diff
@@ -31,10 +31,15 @@ def normalize_interface_headers(header):
 
 
 def parse_network(headers, entry):
+    LIST_OF_STATES = ["ESTABLISHED", "SYN_SENT", "SYN_RECV", "FIN_WAIT1", "FIN_WAIT2", "TIME_WAIT", "CLOSED", "CLOSE_WAIT", "LAST_ACK", "LISTEN", "CLOSING", "UNKNOWN"]
```
Comment (kellyjonbrazil) [inline jc/parsers/netstat_linux.py:34]: Could you add "7" to this list? I haven't seen any other numbers in any other examples and this seems to do the right thing.

**Review by kellyjonbrazil — CHANGES_REQUESTED** [review]
Nice fix - looks like adding "7" to the list fixes a corner case.

**Comment by kellyjonbrazil** [comment]
I'll go ahead and merge and add "7" to the list.

**Comment by HervisDaubeny** [comment]
Hello @kellyjonbrazil , I have found a bug that got to the implementation by adding the `7` to the list of TCP / UDP states. I would like to help you to resolve it, I just want to ask - is it better trying to "reopen" this MergeRequest and related issue or creating a new one?

**Comment by kellyjonbrazil** [comment]
@HervisDaubeny I have created a new issue for this: #578

**Pull request merged**